web3/eth.py

```diff
@@ -158,10 +163,13 @@ def getBlockTransactionCount(self, block_identifier):
             if_hash='eth_getBlockTransactionCountByHash',
             if_number='eth_getBlockTransactionCountByNumber',
         )
-        return self.web3.manager.request_blocking(
-            method,
-            [block_identifier],
-        )
+        try:
+            return self.web3.manager.request_blocking(
+                method,
+                [block_identifier],
+            )
+        except ValueError:
```

Comment:
Does this `ValueError` come from an *error* type response from the JSON-RPC server?  Is this client specific behavior (like geth returns an error response and parity returns empty)?   Struggling to remember how all of this works.

I ask because it feels a little worrisome to be blindly converting `ValueError -> ThingNotFound` as this would be an easy way for our exception messaging to be wrong in a case where the `ValueError` means something else but we're blindly reporting it as a `ThingNotFound`.

Is there a way we can be more precise with this?

Reply:
@pipermerriam The `ValueError` comes from [here](https://github.com/ethereum/web3.py/blob/master/web3/manager.py#L99) which I implemented in the previously-related pr to be raised everytime a jsonrpc call returns `None` - off the top of my head, I don't think it's client-specific behavior meaning that it would probably require some more complexity to have a more precise error message, but if you 👍 I'll dig deeper and see about improving the error reporting

Reply:
I think we should drop at internal raising of the `ValueError` and do the `None` detection within each of the individual methods.  The current `ValueError` based approach has a strong potential for false positives, and completely disallowing `None` returns from `JSON-RPC` calls globally probably isn't a tenable API (sorry I didn't catch that previously).